### PR TITLE
Rel/release 3.8

### PR DESCRIPTION
**Pull request by semani2**
SDK changes to support the following features:
1. Adding/editing meeting providers
2. CRUD reminder information for item
3. CRUD recurrence information for item
4. Repeating calendar events
5. Category coloring of calendar events

### CODE REVIEW

File: sdk/src/androidTest/java/com/podio/sdk/domain/ItemTest.java [hunk 1]
```diff
@@ -27,7 +27,7 @@ public void testAddUnverifiedValue() {
         item.addValue("FIELD-1", new TextField.Value("VALUE"));
 
         Gson gson = new Gson();
-        String actualJson = gson.toJson(item.getCreateData());
+        String actualJson = gson.toJson(item.getCreateData(false));
```
Comment (ken-jen) [inline sdk/src/androidTest/java/com/podio/sdk/domain/ItemTest.java]: What does passing false do here?

Comment (semani2) [inline sdk/src/androidTest/java/com/podio/sdk/domain/ItemTest.java]: I had to update the getCreateData method on the Item object to take an extra parameter called hasLinkedAccountData, if false it uses the default body while making a request or else it adds linked_account parameter to the request's body.

File: sdk/src/main/java/com/podio/sdk/domain/Item.java [hunk 2]
```diff
@@ -335,10 +368,56 @@ public CreateData getCreateData() {
...
+            if(key.equalsIgnoreCase(LinkedAccountDataField.NAME)) {
+                createData.setLinkedAccountId(getLinkedAccountId((LinkedAccountDataField.Value) entry.getValue().get(0)));
+                break;
+            }
+        }
+
+        return createData;
+    }
+
+    private CreateData removeNonFields(CreateData createData) {
```
Comment (ken-jen) [inline sdk/src/main/java/com/podio/sdk/domain/Item.java]: Purpose of removeNonFields?

File: sdk/src/main/java/com/podio/sdk/domain/field/ReminderRecurrenceField.java [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+
+    public ReminderRecurrenceField() {
+        super(NAME);
+        this.values = new ArrayList<>();
+    }
+
+    @Override
+    public void addValue(Value value) {
+        if (values != null && !values.contains(value)) {
+            values.clear();
```
Comment (ken-jen) [inline sdk/src/main/java/com/podio/sdk/domain/field/ReminderRecurrenceField.java]: Do we want to clear all values when adding a single new value?

File: sdk/src/main/java/com/podio/sdk/provider/CalendarProvider.java [hunk 4]
```diff
@@ -87,6 +94,23 @@ public CalendarFilter withTasks(boolean includeTasks) {
     }
 
     /**
+     * Fetches all app calendar events
```
Comment (ken-jen) [inline sdk/src/main/java/com/podio/sdk/provider/CalendarProvider.java]: Do you think we should leave the App Calendar SDK support in?

File: sdk/src/main/java/com/podio/sdk/domain/field/ReminderRecurrenceField.java [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+
+    @Override
+    public void setValues(List<Value> values) {
+        this.values.clear();
+        this.values.addAll(values);
+    }
+
+    @Override
+    public Value getValue(int index) {
+        return values != null ? values.get(index) : null;
```
Comment (semani2) [inline sdk/src/main/java/com/podio/sdk/domain/field/ReminderRecurrenceField.java]: Well, we make sure to set values as soon as we create the values, but I guess it is safe to throw in a index check 👍

Comment (semani2) [inline sdk/src/main/java/com/podio/sdk/domain/field/ReminderRecurrenceField.java]: Fixed it..

File: sdk/src/main/java/com/podio/sdk/provider/CalendarProvider.java [hunk 6]
```diff
@@ -99,9 +93,19 @@ public CalendarFilter withTasks(boolean includeTasks) {
         return get(filter, CalendarEvent[].class);
     }
 
-    public Request<CalendarEvent[]> getAppCalendar() {
-        CalendarFilter filter = new CalendarFilter().withAppId(15850755)
```
Comment (vorgaszeldrahar) [inline sdk/src/main/java/com/podio/sdk/provider/CalendarProvider.java]: Any idea why this appid was hard-coded?

**Pull request merged**